cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/query_enhancements/field_display_filtering.spec.js

```diff
@@ -0,0 +1,150 @@
+  switch (datasetType) {
+    case 'index_pattern':
+      if (language !== 'OpenSearch SQL') {
+        cy.setSearchAbsoluteDateRange(START_TIME, END_TIME);
+      }
+      break;
+  }
+}
+
+function verifyTableFieldFilterActions(datasetType, language, shouldExist) {
+  selectDataset(datasetType, language);
+  setDateRange(datasetType, language);
+
+  cy.getElementByTestId('discoverQueryHits').should('not.exist'); // To ensure it waits until a full table is loaded into the DOM, instead of a bug where table only has 1 hit.
+
+  dataExplorer.verifyDocTableRowFilterForAndOutButton(0, shouldExist);
+
+  if (shouldExist) {
+    dataExplorer.verifyDocTableFilterAction(0, 'filterForValue', '10,000', '1', true);
+    dataExplorer.verifyDocTableFilterAction(0, 'filterOutValue', '10,000', '9,999', false);
+  }
+}
+
+function verifyExpandedTableFilterActions(datasetType, language, isEnabled) {
+  selectDataset(datasetType, language);
+  setDateRange(datasetType, language);
+
+  cy.getElementByTestId('discoverQueryHits').should('not.exist'); // To ensure it waits until a full table is loaded into the DOM, instead of a bug where table only has 1 hit.
+  dataExplorer.toggleDocTableRow(0);
+  dataExplorer.verifyDocTableFirstExpandedFieldFirstRowFilterForFilterOutExistsFilterButtons(
+    isEnabled
+  );
+  dataExplorer.verifyDocTableFirstExpandedFieldFirstRowToggleColumnButtonHasIntendedBehavior();
+
+  if (isEnabled) {
+    dataExplorer.verifyDocTableFirstExpandedFieldFirstRowFilterForButtonFiltersCorrectField(
+      0,
+      0,
+      '10,000',
+      '1'
+    );
+    dataExplorer.verifyDocTableFirstExpandedFieldFirstRowFilterOutButtonFiltersCorrectField(
+      0,
+      0,
+      '10,000',
+      '9,999'
+    );
+    dataExplorer.verifyDocTableFirstExpandedFieldFirstRowExistsFilterButtonFiltersCorrectField(
+      0,
+      0,
+      '10,000',
+      '10,000'
+    );
+  }
+}
+
+describe('filter for value spec', () => {
+  before(() => {
+    // Load test data
+    cy.setupTestData(
+      SECONDARY_ENGINE.url,
+      ['cypress/fixtures/query_enhancements/data-logs-1/data_logs_small_time_1.mapping.json'],
+      ['cypress/fixtures/query_enhancements/data-logs-1/data_logs_small_time_1.data.ndjson']
+    );
+    // Add data source
+    cy.addDataSource({
+      name: `${DATASOURCE_NAME}`,
+      url: `${SECONDARY_ENGINE.url}`,
+      authType: 'no_auth',
+    });
+
+    // Create workspace
+    cy.deleteWorkspaceByName(`${WORKSPACE_NAME}`);
+    miscUtils.visitPage('/app/home');
+    cy.createInitialWorkspaceWithDataSource(`${DATASOURCE_NAME}`, `${WORKSPACE_NAME}`);
+    cy.wait(2000);
+    cy.createWorkspaceIndexPatterns({
+      url: `${BASE_PATH}`,
+      workspaceName: `${WORKSPACE_NAME}`,
+      indexPattern: 'data_logs_small_time_1',
+      timefieldName: 'timestamp',
+      indexPatternHasTimefield: true,
+      dataSource: DATASOURCE_NAME,
+      isEnhancement: true,
+    });
+    cy.navigateToWorkSpaceHomePage(`${BASE_PATH}`, `${WORKSPACE_NAME}`);
```

Comment:
You can move this to the beforeEach block. So that beforeEach test we navigate to the Discover and start from there

Reply:
Would it be better to keep it in before? We are creating a new search in the beforeEach block, navigating to Discover again seems to take extra time for no real benefit.